test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/SystemTest.scala

```diff
@@ -147,4 +148,16 @@ class SystemTest {
       assertEquals(().hashCode(), System.identityHashCode(()))
     }
   }
+
+  @Test def getenv_should_return_an_unmodifiable_map(): Unit = {
+    assertTrue(System.getenv().isInstanceOf[java.util.Map[String, String]])
+
+    expectThrows(classOf[Exception], System.getenv.put("", ""))
+  }
+
+  @Test def getenv_should_link_and_does_not_throw(): Unit = {
```

Comment:
Ah, I didn't realize the JVM case ^^ It's probably better to check something improbable such as `THISVARISNEVERINTHEENV` or something like that, and actually check for `null`.